### ISSUE TITLE
pvc/ns deletion hangs after deletion of a resourceQuota when ResourceQuota Admission Controller is enabled.

### ISSUE THREAD

**Issue by imkin** [kubernetes/kubernetes#75930]
**What happened**:

ResourceQuota Admission Controller is enabled and configured.
While cleaning up the Namespace sometimes deletion hangs indefinitely. 
Found that the deletion was hanging because pvc is indefinitely in terminal state.

**What you expected to happen**:

namespace and pvc cleaned up successfully.

**How to reproduce it (as minimally and precisely as possible)**:

k8s version 1.14.0 (on gcp) (reproducible in older versions too but for these steps indulge me).
apiserver configured with admission-controller-config-file

```
#admission_control.conf
kind: AdmissionConfiguration
apiVersion: apiserver.k8s.io/v1alpha1
plugins:
- name: "ResourceQuota"
  configuration:
    apiVersion: resourcequota.admission.k8s.io/v1beta1
    kind: Configuration
    limitedResources:
    - resource: persistentvolumeclaims
      matchContains:
      - .storageclass.storage.k8s.io/requests.storage
```
```
# sc.yaml
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: test-profile
provisioner: kubernetes.io/gce-pd
parameters:
  type: pd-standard
```
```
# pvc.yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  namespace: us1-dev-web-1 
  name: test-pvc
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 2Gi
  storageClassName: test-profile
```
```
# rq.yaml
apiVersion: v1
kind: ResourceQuota
metadata:
  namespace: us1-dev-web-1
  name: us1-dev-web-1-storagequota
spec:
  hard:
    test-profile.storageclass.storage.k8s.io/requests.storage: 90Gi
```
 Step 1: create all the objects. 
```
$ kubectl apply -f sc.yaml 
$ kubectl apply -f ns.yaml
$ kubectl apply -f rq.yaml
$ kubectl apply -f pvc.yaml
```

Step 2: delete resource quota

```
$ kubectl delete -f rq.yaml
```

Step 3: delete the namespace

```
$ kubectl delete -f ns.yaml
```

```
$ kubectl get ns us1-dev-web-1
NAME            STATUS        AGE
us1-dev-web-1   Terminating   18h

$ kubectl get pvc -n us1-dev-web-1
NAME       STATUS        VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
test-pvc   Terminating   pvc-19938bd6-541c-11e9-b978-42010a8a0022   2Gi        RWO            test-profile   160m
```
This leaves the ns and pvc in terminating state.

**Anything else we need to know?**:

This leaves the namespace is non-recoverable state. 

If you do not delete the rq.yaml in Step 2 above and jump to Step 3 it is possible while deleting ns there is race condition and the resource quota could be delete before the pvc.

**Environment**:
- Kubernetes version (use `kubectl version`): reproduced in 1.14.0, 1.13.0, 1.12.0
- Cloud provider or hardware configuration: gcp (should not matter though)
- OS (e.g: `cat /etc/os-release`): cos (shoudl not matter
- Kernel (e.g. `uname -a`): NA
- Install tools: NA
- Others: NA

**Comment by imkin** [comment]
The controller-manager logs show

```
pvc_protection_controller.go:138] PVC us1-dev-web-1/test-pvc failed with : persistentvolumeclaims 
"test-pvc" is forbidden: insufficient quota to consume: test-
profile.storageclass.storage.k8s.io/requests.storage
```

**Comment by mansi-a** [comment]
I'm working on a fix for this. Please wait for my pull request.

**Comment by imkin** [comment]
Thanks @mansi-a for looking into this. 
/assign @mansi-a

**Comment by imkin** [comment]
@mansi-a ignore the robot. Please share your PR here and that should suffice.

**Comment by imkin** [comment]
Looks like #76299 was merged. Waiting for #77367 and #77375 to merge and then will close this. 
@mansi-a thanks a lot for working on this one.

**Comment by imkin** [comment]
/close 

I am good with the changes already done. The https://github.com/kubernetes/kubernetes/pull/76299 fix would be part of 1.15 and https://github.com/kubernetes/kubernetes/pull/77367 would be cherry-pickable for 1.14. 
https://github.com/kubernetes/kubernetes/pull/77375 would fix 1.13 and 1.12 is on it way out soon.
